importer/mcp/fetcher.go

```diff
@@ -0,0 +1,196 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"slices"
+	"strconv"
+	"time"
+
+	mcpapiv0 "github.com/modelcontextprotocol/registry/pkg/api/v0"
+)
+
+const (
+	// defaultPageLimit is the default number of servers to fetch per page.
+	defaultPageLimit = 30
+)
+
+// Supported filters https://registry.modelcontextprotocol.io/docs#/operations/list-servers#Query-Parameters
+//   - search: Filter by server name (substring match)
+//   - version: Filter by version ('latest' for latest version, or an exact version like '1.2.3')
+//   - updated_since: Filter by updated time (RFC3339 datetime)
+//   - limit: Number of servers per page (default 30)
+//   - cursor: Pagination cursor
+var supportedFilters = []string{
+	"search",
+	"version",
+	"updated_since",
+	"limit",
+	"cursor",
+}
+
+// Fetcher implements the pipeline.Fetcher interface for MCP registry.
+type Fetcher struct {
+	baseURL    string
+	httpClient *http.Client
+	filters    map[string]string
+	limit      int
+}
+
+// NewFetcher creates a new MCP fetcher.
+func NewFetcher(baseURL string, filters map[string]string, limit int) *Fetcher {
+	return &Fetcher{
+		baseURL: baseURL,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second, //nolint:mnd
+		},
+		filters: filters,
+		limit:   limit,
+	}
+}
+
+// Fetch retrieves servers from the MCP registry and sends them to the output channel.
+func (f *Fetcher) Fetch(ctx context.Context) (<-chan interface{}, <-chan error) {
+	outputCh := make(chan interface{})
```

Comment:
```suggestion
    // Use buffered channel to allow fetcher to work ahead of transformers
	outputCh := make(chan interface{}, 50)
```
A buffered channel has internal storage for N items. Sends only block when the buffer is full. This is a minimal backpressure control. We can define a constant for the buffer size.